android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/DirectDispatcher.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2023. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.rib.core
+
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineDispatcher
+
+/**
+ * A dispatcher that immediately executes the [Runnable] on the same stack frame, without
+ * potentially forming event loops like [Unconfined][kotlinx.coroutines.Dispatchers.Unconfined] or
+ * [Main.immediate][kotlinx.coroutines.MainCoroutineDispatcher.immediate] in case of nested
+ * coroutines.
+ *
+ * For more context, see the following issues on `kotlinx.coroutines` GitHub repository:
+ * <!-- spotless:off -->
+ * 1. [Immediate dispatchers can cause spooky action at a distance](https://github.com/Kotlin/kotlinx.coroutines/issues/3760)
+ * 2. [Chaining rxSingle calls that use Unconfined dispatcher and blockingGet results in deadlock #3458](https://github.com/Kotlin/kotlinx.coroutines/issues/3458)
+ * 3. [Coroutines/Flow vs add/remove listener (synchronous execution) #3506](https://github.com/Kotlin/kotlinx.coroutines/issues/3506)
+ * <!-- spotless:on -->
+ */
+internal object DirectDispatcher : CoroutineDispatcher() {
```

Comment:
thanks for adding! wondering if this should be moved into another module (e.g. libraries/rib-coroutines)

Reply:
@psteiger friendly ping here ^

Reply:
Sorry I missed this comment.

I think it's best to have it in "core", because I believe the tendency is for rib-coroutine to cease to exist once we migrate fully to coroutines.

@tyvsmith in case you wanna confirm or deny ^

Reply:
np! I see, thanks for clarifying

Reply:
Lets move it to ribs-coroutines until we're ready to move all of it to core. Otherwise we're in a worse fragmented state.